sourcegraph/scip-ruby
# chore: Merge upstream master into scip-ruby/master.

- Use enum instead of a bool for the typechecking path (#6452)
- Update test runner to support more than 2 typechecking paths (#6453)
- Add type signatures for Tempfile (#6455)
- Make runtime type check `attr_accessor` writers (#6426)
- Fix return type for IO#read without a length (#6464)
- Ignore formatting requests on __package.rb files (#6465)
- Bring back package-awareness to Sorbet-autogen (#6466)
- Add symbol-table-exp test (#6467)
- Some no-op / pre-work edits (#6470)
- add more logging for `SorbetException` in `main` (#6458)
- delete `FileReadException` (#6473)
- 🔥 Delete LSP stale state mode (#6472)
- Don't mangle constant symbols (#6471)
- staticField.name -> name (#6476)
- make `FileNotFoundException` have an actual message (#6474)
- Also silence `private` method redefinition warning (#6479)
- fix `test/cli/bad-perm` output (#6480)
- rearrange `appendFilesInDir` to do faster early rejects (#6482)
- Generate package-based autoload shims for path-based autoloading (#6478)
- Update docs for `T.must_because` (#6481)
- don't allocate strings in `PrinterConfig::flush` (#6490)
- don't copy `AutogenResult::Serialized` in workers (#6489)
- copy less stuff when using `msgpack` in autogen (#6495)
- only look up the entry in `childMap` once per iteration (#6496)
- fix `vscode_extension/README.md` text (#6497)
- update `vscode-languageclient` with LSP 3.17 support (#6492)
- Fix crash in getCallArguments (#6484)
- Add error for non-`Proc`-typed block argument (#6486)
- Implement `Array#dig` and `Hash#dig` intrinsic (#6491)
- delete unused `blockRetrunTemp` (#6500)
- delete some dead forward definitions from `shared_ptr<core::Type>` days (#6499)
- Revert "update vscode-languageclient with LSP 3.17 support (#6492)" (#6501)
- Revert "Make runtime type check `attr_accessor` writers (#6426)" (#6505)
- Avoid excess allocations in `T::Types::Enum#name` (#6506)
- Skip checking block type in sorbet-runtime when possible (#6487)
- Update T::Struct docs (#6502)
- Add IRB::RelineInputMethod RBI definition (#6509)
- Fix accidental typo in test name (#6510)
- Prevent C/C++ Extension Pack recommendation (#6513)
- Docs: missing word (#6515)
- Empty --autogen-autoloader-pbal-namespaces registers all strict packages for PBAL by default (#6511)
- we use C++17 now, enforce it in the code (#6525)
- eliminate the `spd` namespace alias (#6524)
- Pre-resolve ::<PackageSpecRegistry> in __package.rb rewriter (#6518)
- Remove call to `dealias` in namer (#6519)
- move includes to `WorkerPoolImpl.cc` when possible (#6526)
- ENFORCE that owner.kind() == Kind::Class (#6527)
- chore: Bump mentioned master commit.

### Motivation

Sync before making more progress

### Test plan

Covered by existing tests.
